src/formats/frontmatter.js

```diff
@@ -1,16 +1,43 @@
-import preliminaries from 'preliminaries';
-import yamlParser from 'preliminaries-parser-yaml';
-import tomlParser from 'preliminaries-parser-toml';
+import matter from 'gray-matter';
+import tomlEng from 'toml';
 import YAML from './yaml';
 
-// Automatically register parsers
-preliminaries(true);
-yamlParser(true);
-tomlParser(true);
+const parsers = {
+  toml: tomlEng.parse.bind(tomlEng),
+  json: (input) => {
+    let JSONinput = input.trim();
+    // Fix JSON if leading and trailing brackets were trimmed.
```

Comment:
Why would brackets be trimmed?

Reply:
Delimiters always get trimmed. For stuff like YAML where the delims are `---` and not actually part of the code that is helpful. But with JSON, the delimiters `{` and `}` get trimmed, but they actually need to be in there for the parser. With `preliminaries` they did this too, here: [preliminaries/index.js#L376-L379](https://github.com/josephearl/preliminaries/blob/f7844b1ed9c2237374cce64510aba2a829993961/preliminaries/index.js#L376-L379)

We can't just add them all the time, though, we have to `if`-check first, because they could have wrapped the whole thing in a `---json` delimiter, in which case the brackets wouldn't get trimmed.

Reply:
You can also see the difference in the tests:
https://github.com/netlify/netlify-cms/blob/3195f30b8d174b4820ef65d07d64ded336477af8/src/formats/__tests__/frontmatter.spec.js#L69
vs
https://github.com/netlify/netlify-cms/blob/3195f30b8d174b4820ef65d07d64ded336477af8/src/formats/__tests__/frontmatter.spec.js#L81